README.md

```diff
@@ -15,8 +15,80 @@ following entitlements are supported:
+| Contract Server | The backend service exposing a REST API to which UA Client authenticates in order to obtain contract and commercial service information and manage which support servics are active on a machine.|
+| Entitlement/Service | An Ubuntu Advantage commercial support service such as FIPS, ESM, Livepatch, CIS-Audit to which a Contract may be entitled |
+| Affordance | Service-specific list of applicable architectures and Ubuntu series on which a service can run |
+| Directives | Service-specific configuration values which are applied to a service when enabling that service |
+| Obligations | Service-specific policies that must be instrumented for support of a service. Example: `enableByDefault: true` means that any attached machine **MUST** enable a service on attach |
+
+
+## Architecture
+Ubuntu Advantage client, hereafter "UA client", is python3-based command line utility. It provides a CLI to attach, detach, enable,
+disable and check status of support related services.
+
+The package `ubuntu-advantage-tools` also provides a C++ APT hook which helps advertise ESM service and available packages in MOTD and during various apt commands.
+
+The `ubuntu-advantage-pro` package delivers auto-attach auto-enable functionality via init scripts and systemd services for various cloud platforms.
+
+By default, Ubuntu machines are deployed in an unattached state. A machine can get manually or automatically attached to a specific contract by interacting with the Contract Server REST API to enable a machine. Any change in state of services or machinbe attach results in additional interactions with the Contract Server API to validate such operations. The contract server API is described by the [ua-contracts openapi spec](https://github.com/CanonicalLtd/ua-contracts/blob/develop/docs/contracts.yaml).
+
+### Attaching a machine
+Each Ubuntu SSO account holder has access to one one or more contracts. To attach a machine to an Ubuntu Advantage contract:
+
+* An Ubuntu SSO account holder must obtain a contract token from https://ubuntu.com/advantage.
+* Run `ua attach <contractToken>` on the machine or provide comparable
+* UA Client reads config from /etc/ubuntu-advantage/uaclient.conf nad  a request containing the `<contractToken>` to the Contract Server API @ https://<contract_url>/api/v1/context/machines/token and obtains an unique machine-token JSON response containing credentials, affordances, directives and obligations to allow enabling and disabling any Ubuntu Advantage service
+* UA Client POSTs a request containing the `<contractToken>` to the Contract Server API @ https://contracts.canonical.com/api/v1/context/machines/tokento to obtain an unique machine-token JSON response which contains credentials, affordances, directives and obligations to allow enabling and disabling any Ubuntu Advantage service
```

Comment:
I think you can remove the ending `to` from the url `https://contracts.canonical.com/api/v1/context/machines/tokento` 

Also. maybe I am missing something, but this paragraph seems really similar to the last one. I believe they are different ways to obtain the same JSON response and if that is the case, maybe we could address it in the docs. But it possible that | did not understand it correctly

Reply:
@lucasmoura Sorry I had accidentally double added that content. I've restructure the text and alignment and cleaned up a lot of inconsistencies now.